src/seventh/client/weapon/ClientWeapon.java

```diff
@@ -253,6 +230,31 @@ public void update(TimeStep timeStep) {
             }
         }
     }
+
+	private void firingUpdate() {
+		firstFire++;
+		if(!onFire()) {           
+		    if(canFireAgain) {                                    
+		        canFireAgain = false;
+		    }
+		    else {
```

Comment:
If you are not going to implement the logic for `else` case, then should not let it appears empty like that.